### PR TITLE
Collapsible/CollapsibleItem Fixes

### PR DESCRIPTION
**Pull request by alextrastero**

### CODE REVIEW

**Comment by alextrastero** [comment]
Fixes #80

File: src/CollapsibleItem.js [hunk 1]
```diff
@@ -1,86 +1,31 @@
...
+  header,
+  children,
+  icon,
+  iconClassName,
+  node: Node,
+  onSelect,
+  ...props
+}) => (
+  <li className={cx(className, { active: expanded })} {...props}>
+    <Node
```
Comment (isaiah) [inline src/CollapsibleItem.js]: here should be `node`.

File: src/CollapsibleItem.js [hunk 2]
```diff
@@ -1,86 +1,31 @@
...
+      className={cx('collapsible-header', { active: expanded })}
+      onClick={() => onSelect(eventKey)}
```
Comment (isaiah) [inline src/CollapsibleItem.js]: we still need the default `handleClick` function to wrap the custom `onSelect` callback, and handle the toggle of `expanded` state. Overall this cannot be a stateless component, the `expanded` state has to be hold.

File: src/CollapsibleItem.js (hunk 1 above)
Comment (alextrastero) [inline src/CollapsibleItem.js]: `node` is renamed to `Node` in the deconstruct

File: src/CollapsibleItem.js (hunk 2 above)
Comment (alextrastero) [inline src/CollapsibleItem.js]: this is handled by parent component `Collapsible` L56 no?

Comment (isaiah) [inline src/CollapsibleItem.js]: Hmm, I'm not sure any more, can you try on the docs page and see if it work?

**Comment by alextrastero** [comment]
Nah, it's not working well, checking further

**Comment by alextrastero** [comment]
ping @isaiah works like a charm now :)

**Comment by isaiah** [comment]
@alextrastero there is a failing spec.

**Comment by alextrastero** [comment]
because it's not working like a charm... :(

**Comment by alextrastero** [comment]
@mattBlackDesign would you mind testing the new changes?

**Comment by mattBlackDesign** [comment]
Hey @alextrastero I noticed some issues with `defaultActiveKey`. 

When `Collapsible` takes the following form it works as expected:

```
<Collapsible accordion popout defaultActiveKey={0}>         
  <CollapsibleItem header='First'>A</CollapsibleItem>
  <CollapsibleItem header='Second'>B</CollapsibleItem>
</Collapsible>
```

However, when `Collapsible` has children inside of it that are components that render `CollapsibleItem`, `defaultActiveKey` does not expand the specific element. 

i.e. 

```
<Collapsible accordion defaultActiveKey={0}>
  <ChildComponentWithCollapsibleItem />
  <ChildComponentWithCollapsibleItem />
</Collapsible>

...

class ChildComponentWithCollapsibleItem extends Component {
  ...
  render () {
    return(
      <CollapsibleItem header='Element'>Element</CollapsibleItem>
    )
  }
}
```

**Comment by alextrastero** [comment]
In this case `ChildComponentWithCollapsibleItem` should pass all props to children:
```js
class ChildComponentWithCollapsibleItem extends Component {
  ...
  render () {
    const { children, {...other } = this.props;
    return(
      <CollapsibleItem header='Element' {...other}>{children}</CollapsibleItem>
    )
  }
}
```

**Pull request merged**